lib/Sema/TypeCheckType.cpp

```diff
@@ -920,6 +920,10 @@ static Type applyGenericArguments(Type type, TypeResolution resolution,
       auto arg = found->rhs;
       if (auto *expansionType = arg->getAs<PackExpansionType>())
         arg = expansionType->getPatternType();
+
+      if (arg->isParameterPack())
```

Comment:
PackMatcher should be refactored to do this. Right now it _unwraps_ one element packs, and here we re-wrap them